plugins/modules/ssh_config.py

```diff
@@ -274,6 +283,15 @@ def ensure_state(self):
             controlpersist=fix_bool_str(self.params.get('controlpersist')),
             dynamicforward=self.params.get('dynamicforward'),
         )
+        if self.params.get('other_options'):
+            for key, value in self.params.get('other_options').items():
+                if key not in args:
+                    if not isinstance(value, string_types):
```

Comment:
Why force the input to be a string? Wouldn't be simpler (and more friendly to the user) to use `str(value)`?

Reply:
The code previously did this at [1d42d1d](https://github.com/ansible-collections/community.general/pull/9684/commits/1d42d1d018283550f023f316994d5a357c07c916) but was changed to check type and fail instead as result of review above...

Reply:
Thanks for pointing out :-)

Reply:
Conversion to string is generally dangerous, think of `True`/`False` (could have been a `yes`/`no`/`true`/`false`/...), or `1.10` (float that's converted to `"1.0"` when stringified). Or a JSON object that's passed as a `dict` (because ansible-core decided to parse it) and then converted into Python dict format, which isn't JSON. Such wrong stringifications can take users a long time to debug.